torchchat/utils/quantize.py

```diff
@@ -110,23 +125,73 @@ def quantize_model(
 
     if isinstance(quantize_options, str):
         quantize_options = json.loads(quantize_options)
-
     for quantizer, q_kwargs in quantize_options.items():
         if quantizer not in quantizer_class_dict:
             raise RuntimeError(f"unknown quantizer {quantizer} specified")
         else:
             # Use tensor subclass API for int4 weight only.
```

Comment:
Whoops, Comment got split off from:`if (device in ["cuda", "xpu", "npu"]) and quantizer == "linear:int4":`